assets/src/blocks/Submenu/getHeadingsFromBlocks.js

```diff
@@ -0,0 +1,77 @@
+import { generateAnchor } from './generateAnchor';
+
+// We can put the other blocks that can have a heading inside in here along with the attribute containing the heading text.
+// Then we can also filter those to include them in the menu.
+const blockTypesWithHeadings = [
+  {name: 'planet4-blocks/articles', fieldName: 'article_heading', level: 2},
+];
+
+export const getHeadingsFromBlocks = (blocks, selectedLevels) => {
+  const headings = [];
+  blocks.forEach(block => {
+    if (block.name === 'core/heading') {
+      const blockLevel = block.attributes.level;
+
+      const levelConfig = selectedLevels.find(selected => selected.heading === blockLevel);
+
+      if (!levelConfig) {
+        return;
+      }
+
+      const anchor = block.attributes.anchor || generateAnchor(block.attributes.content, headings.map(h => h.anchor));
+
+      headings.push({
+        level: blockLevel,
+        content: block.attributes.content,
+        anchor,
+        style: levelConfig.style,
+        shouldLink: levelConfig.link,
+      });
+
+      return;
+    }
+
+    if (block.name === 'core/freeform') {
+      const parser = new DOMParser();
+      const selector = selectedLevels.map(({heading})=>`h${heading}`).join();
+      const doc = parser.parseFromString(block.attributes.content, 'text/html');
+
+      const classicHeadings = doc.querySelectorAll(selector);
+
+      headings.push(...[...classicHeadings].map(h => {
```

Comment:
Is the leftmost spread really needed here? 🤔

Reply:
Yes, `classHeadings` is a `NodeList` which doesn't have map, so the inner one makes it an array, then the outer one spreads that array into multiple arguments of push, else it would push the array itself as a single element.